fiftyone/utils/eval/openimages.py

```diff
@@ -0,0 +1,699 @@
+            predicted detections and labels according to the provided 
+            hierarchy
+    """
+
+    def __init__(
+        self,
+        pred_field,
+        gt_field,
+        iou=None,
+        classwise=None,
+        iscrowd="IsGroupOf",
+        max_preds=None,
+        hierarchy=None,
+        pos_lab_field=None,
+        neg_lab_field=None,
+        expand_hierarchy=True,
+        expand_pred_hierarchy=False,
+        **kwargs
+    ):
+        super().__init__(
+            pred_field, gt_field, iou=iou, classwise=classwise, **kwargs
+        )
+
+        if max_preds is None:
+            max_preds = 100
+
+        self.iscrowd = iscrowd
+        self.max_preds = max_preds
+        self.hierarchy = hierarchy
+        self.pos_lab_field = pos_lab_field
+        self.neg_lab_field = neg_lab_field
+        self.expand_hierarchy = expand_hierarchy
+        self.expand_pred_hierarchy = expand_pred_hierarchy
+
+        if expand_pred_hierarchy:
+            if not hierarchy:
+                logger.info(
+                    "No hierarchy provided, setting expand_pred_hierarchy to False"
+                )
+                self.expand_pred_hierarchy = False
+
+            # If expand pred hierarchy is true, so is expand hierarchy
+            self.expand_hierarchy = self.expand_pred_hierarchy
+
+        if expand_hierarchy and not hierarchy:
+            logger.info(
+                "No hierarchy provided, setting expand_hierarchy to False"
+            )
+            self.expand_hierarchy = False
+
+        if self.hierarchy:
+            (
+                self.hierarchy_keyed_parent,
+                self.hierarchy_keyed_child,
+                _,
+            ) = self._build_plain_hierarchy(self.hierarchy, skip_root=True)
+
+    @property
+    def method(self):
+        return "openimages"
+
+    @property
+    def additional_fields_required(self):
+        return True
+
+    # Parse hierarchy, code from:
+    # https://github.com/tensorflow/models/blob/ec48284d0db7a67ab48a9bc13dc29c643ce0f197/research/object_detection/dataset_tools/oid_hierarchical_labels_expansion.py#L77
+    def _update_dict(self, initial_dict, update):
+        """Updates dictionary with update content.
+
+        Args:
+            initial_dict: initial dictionary.
+            update: updated dictionary.
+        """
+
+        for key, value_list in update.items():
+            if key in initial_dict:
+                initial_dict[key].update(value_list)
+            else:
+                initial_dict[key] = set(value_list)
+
+    def _build_plain_hierarchy(self, hierarchy, skip_root=False):
+        """Expands tree hierarchy representation to parent-child dictionary.
+
+        Args:
+            hierarchy: labels hierarchy .
+            skip_root: if true skips root from the processing (done for the case when all
+                classes under hierarchy are collected under virtual node).
+        Returns:
```

Comment:
nit: remove periods from args (for consistency with rest of codebase) and and newline before `Returns`